internal/samsahai/controller.go

```diff
@@ -1088,6 +1090,33 @@ func (c *controller) GetActivePromotionHistory(name string) (v *s2hv1beta1.Activ
 	return
 }
 
+func (c *controller) GetDeployEngine(teamName, ns string) internal.DeployEngine {
```

Comment:
can we rename it to be like `GetActivePromotionDeployEngine` since this function returns the deployment engine of active promotion only?